src/main/java/omoh/tasktypes/Task.java

```diff
@@ -0,0 +1,97 @@
+package omoh.tasktypes;
+import omoh.Omoh;
+
+import omoh.customexceptions.EmptyTaskNumberException;
+
+public class Task {
+    protected static Task[] tasks;
+    public static int totalTasks = 0;
+    protected String description;
+    protected boolean isDone;
+
+    protected String type;
+
+    public static void initArray() {
+        tasks = new Task[100];
+    }
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+        this.type = "T";
+    }
+
+
+    public static void getAllTasks() {
+        System.out.println("Here are the tasks in your list:");
+        int serialNumber = 1;
+        for (int i = 0; i < totalTasks; i++) {
+            System.out.println(serialNumber + "." + tasks[i].toString());
+            serialNumber += 1;
+        }
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    //method that extracts the task number to mark or unmark
+    public static int extractTaskNumber(String input) throws NumberFormatException, EmptyTaskNumberException {
+        String keyword;
+        if (input.startsWith("mark")) {
+            keyword = "mark";
+        } else {
+            keyword = "unmark";
+        }
+        String numberString = input.substring(keyword.length()).trim();
+        if (numberString.isEmpty()) {
+            throw new EmptyTaskNumberException();
+        }
+        int taskNumber = Integer.parseInt(numberString);
+        return taskNumber;
+    }
+
+
+    //method that modifies whether task is done or not done, depending on keyword mark or unmark detected
+    public static void modifyDoneState(int taskNumber, String input) {
+        //only executes if taskNumber is valid
+        if (taskNumber != -1) {
```

Comment:
You could possibly refactor this into a guard clause to avoid deep nesting! Think about how you can apply the idea of guard clauses thoughout your codebase to make the [happy path more prominent](https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent).

Also, there are many magic literals throughout the codebase. Do refactor them! For example, it is not entirely clear what `-1` means here.